### PR TITLE
create BeetrootFinance adapter

### PR DESCRIPTION
**Pull request by welaskez**
added adapter for the BeetrootFinance project

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/BeetrootFinance exports TVL: 
        
 
 
    ton                       0
    
    total                    0

**Comment by welaskez** [comment]
> please compute tvl using on chain data and please fill in project details

everything is counted onchain through deposits and withdrawals and where you have to fill in project details?

File: projects/BeetrootFinance/index.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
+const { get } = require('../helper/http')
+const ADDRESSES = require("../helper/coreAssets.json");
+
+async function tvl(api) {
+  const data = await get('https://api.beetroot.finance/v1/tvl')
```
Comment (g1nt0ki) [inline projects/BeetrootFinance/index.js:6]: no, you are reading from your api

**Comment by g1nt0ki** [comment]
> > please compute tvl using on chain data and please fill in project details
> 
> everything is counted onchain through deposits and withdrawals and where you have to fill in project details?

it should be in the PR description like https://github.com/DefiLlama/DefiLlama-Adapters/pull/12350

**Comment by welaskez** [comment]
> please compute tvl using on chain data and please fill in project details

**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Name (to be shown on DefiLlama): BeetrootFinance


##### Twitter Link: https://x.com/beetroot_fi


##### List of audit links if any:


##### Website Link:


##### Logo (High resolution, will be shown with rounded borders): https://raw.githubusercontent.com/Beetroot-fi/contracts/refs/heads/master/LOGO.PNG



##### Current TVL: 0


##### Treasury Addresses (if the protocol has treasury)


##### Chain: TON Blockchain (The Open Network)


##### Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): (https://api.coingecko.com/api/v3/coins/list)


##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap, leave empty if not listed): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)


##### Short Description (to be shown on DefiLlama): BeetrootFinance is an automated yield farming aggregator, supply $USDT and get best APY.


##### Token address and ticker if any: ROOT: https://tonviewer.com/EQAFGhmx199oH6kmL78PGBHyAx4d5CiJdfXwSjDK5F5IFyfC


##### Category (full list at https://defillama.com/categories) *Please choose only one: Yield Aggregator


##### Oracle Provider(s): Specify the oracle(s) used (e.g., Chainlink, Band, API3, TWAP, etc.):
##### Implementation Details: Briefly describe how the oracle is integrated into your project:
##### Documentation/Proof: Provide links to documentation or any other resources that verify the oracle's usage:

##### forkedFrom (Does your project originate from another project):


##### methodology (what is being counted as tvl, how is tvl being calculated): TVL calculation methodology consists of the delta between onchain USDT deposits and withdrawals


##### Github org/user (Optional, if your code is open source, we can track activity): https://github.com/Beetroot-fi

**Comment by llamatester** [comment]
The adapter at projects/BeetrootFinance exports TVL: 
        
 
 
    ton                       0
    
    total                    1.0000133297077001e-12

**Comment by llamatester** [comment]
The adapter at projects/BeetrootFinance exports TVL: 
        
 
 
    ton                       0
    
    total                    1.0000133297077001e-12

**Comment by llamatester** [comment]
The adapter at projects/BeetrootFinance exports TVL: 
        
 
 
    ton                       1
    
    total                    0.9979988298641247

**Pull request merged**